### ISSUE TITLE
Graphing: provide a matplotlib backend that plots to a browser

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#3668]
As an instrument scientist, I would like the option of a matplotlib backend that plots to a browser in a non-blocking way (rather than plotting into IBEX).

This is an easy extension of the backend that we've already done (which already does the non-blocking part).

An example use case for this (from LARMOR) is:
- I want my IBEX client pointing at the DAE perspective so that I can see spectra etc
- I am running my script in a standalone python window
- I want to be able to have non-blocking plots, entirely separately from the IBEX client.

**Comment by ChrisM-S** [comment]
Does this work off the instrument on a remote system with only Genie Python installed? i.e. providing remote DAE plots in GENIE python (granted that IBEX/ICP are running to serve data).

Alternatively, is this equivalent to the remote Open GENIE DAE spectra plot?

**Comment by Tom-Willemsen** [comment]
> Does this work off the instrument on a remote system with only Genie Python installed? 

Yes. It is all self-contained. The current version already does this actually - if it can't plot into IBEX then it will fall back to giving you a link to go to in a web browser. This ticket is just to let the user explicitly choose that behaviour, even if IBEX is open.

> i.e. providing remote DAE plots in GENIE python (granted that IBEX/ICP are running to serve data).

This is just the plotting layer - getting the data is completely seperate.

> Alternatively, is this equivalent to the remote Open GENIE DAE spectra plot?

I'm not sure how open genie worked but this is just a plotting layer, it is not tied to spectra plots in any way.
